app/jobs/bulkrax/schedule_relationships_job_decorator.rb

```diff
@@ -7,14 +7,12 @@ module ScheduleRelationshipsJobDecorator
     def perform(importer_id:)
       importer = Importer.find(importer_id)
 
-      processed_entries =
-        Bulkrax::Status
-        .where(statusable_id: importer.entries.pluck(:id))
-        .where(runnable_id: importer.last_run.id)
-        .where.not(status_message: 'Pending')
-        .pluck(:statusable_id).uniq
-      pending_entries = importer.entries.where.not(id: processed_entries)
-      pending_num = pending_entries.count
+      # If we don't have anymore unfinished ImportWorkJob jobs, then we
+      # can be confident that we are ready to create the relationships
+      pending_num = ::GoodJob::Job.where("serialized_params->>'job_class' = ?", 'Bulkrax::ImportWorkJob')
+                                  .where(finished_at: nil)
+                                  .where("(serialized_params->'arguments'->1)::text = ?", importer.last_run.id.to_s)
+                                  .count
```

Comment:
> If we don't have anymore unfinished ImportWorkJob jobs, then we can be confident that we are ready to create the relationships

I don't think this is accurate. The import work job starts the process of creating a work, but the work doesn't necessarily exist by the time the import work job is done (since the creation is queued up as another job). Keying off of the jobs leaves room for race conditions compared to the status, which isn't set until the work either succeeds or fails to create